debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/ShrCCommand.java

```diff
@@ -1042,167 +1134,195 @@ private void dbgShrcPrintAllStats(PrintStream out, J9JavaVMPointer vm, J9SharedC
 
-		public J9SharedCacheHeaderPointer getCacheStartAddress() {
+		public J9SharedCacheHeaderPointer[] getCacheStartAddress() {
 			return cacheStartAddress;
 		}
 
-		public UDATA getRomclassStartAddress() {
+		public UDATA[] getRomclassStartAddress() {
 			return romclassStartAddress;
 		}
 
-		public UDATA getSegmentPtr() {
+		public UDATA[] getSegmentPtr() {
 			return segmentPtr;
 		}
 	}
 	
-	U8Pointer getSharedCacheMetadataStart(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException {
+	U8Pointer[] getSharedCacheMetadataStart(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException {
 		J9SharedClassConfigPointer sharedConfig = vm.sharedClassConfig();
 		if (sharedConfig.isNull()) {
-			CommandUtils.dbgPrint(out,"SharedClassConfig can not be found.");
+			CommandUtils.dbgPrint(out, "SharedClassConfig can not be found.\n");
 			return null;
 		}
 		
 		ShrcConfig config = dbgShrcReadConfig(sharedConfig, out);
-		J9SharedCacheHeaderPointer header = config.getCacheStartAddress();
-		J9SharedCacheHeaderInfo helper = new J9SharedCacheHeaderInfo(header);
-		return U8Pointer.cast(helper.getMetaDataEnd());
+		J9SharedCacheHeaderPointer[] header = config.getCacheStartAddress();
+
+		U8Pointer[] ret = new U8Pointer[header.length];
+		for (int i = 0; i < header.length; i++) {
+			J9SharedCacheHeaderInfo helper = new J9SharedCacheHeaderInfo(header[i]);
+			ret[i] = U8Pointer.cast(helper.getMetaDataEnd());
+		}
+		return ret;
 	}
 	
-	U8Pointer getSharedCacheMetadataEnd(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException {
+	U8Pointer[] getSharedCacheMetadataEnd(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException {
 		J9SharedClassConfigPointer sharedConfig = vm.sharedClassConfig();
 		if (sharedConfig.isNull()) {
-			CommandUtils.dbgPrint(out,"SharedClassConfig can not be found.");
+			CommandUtils.dbgPrint(out, "SharedClassConfig can not be found.\n");
 			return null;
 		}
 		
 		ShrcConfig config = dbgShrcReadConfig(sharedConfig, out);
-		J9SharedCacheHeaderPointer header = config.getCacheStartAddress();
-		J9SharedCacheHeaderInfo helper = new J9SharedCacheHeaderInfo(header);
-		return U8Pointer.cast(helper.getMetaDataStart().longValue() - ShcItemHdr.SIZEOF);
-	}
+		J9SharedCacheHeaderPointer[] header = config.getCacheStartAddress();
 
-	SharedCacheMetadata dbgReadSharedCacheMetadata(J9JavaVMPointer vm, PrintStream out) throws CorruptDataException {
-		U8Pointer heapTop = getSharedCacheMetadataEnd(vm, out);
-		U8Pointer heapBase = getSharedCacheMetadataStart(vm, out);
-		
-		if (heapTop.isNull() || heapBase.isNull()) {
-			CommandUtils.dbgPrint(out, "SharedCache is not initialized enough to find the metadata area of the cache");
-			return null;
+		U8Pointer[] ret = new U8Pointer[header.length];
+		for (int i = 0; i < header.length; i++) {
+			J9SharedCacheHeaderInfo helper = new J9SharedCacheHeaderInfo(header[i]);
+			ret[i] = U8Pointer.cast(helper.getMetaDataStart().longValue() - ShcItemHdr.SIZEOF);
```

Comment:
`longValue()` will throw an exception for values larger than `Long.MAX_VALUE`; this must be written as
```
U8Pointer.cast(helper.getMetaDataStart()).sub(ShcItemHdr.SIZEOF)
```

Reply:
Fixed